### ISSUE TITLE
Homepage: section headers are not read by screen reader

### ISSUE THREAD

**Issue by danirabbit** [elementary/appcenter#2302]
### What Happened?

When navigating the home page via screen reader, section headers (like recently added or categories) aren't read. 

### Steps to Reproduce

1. Turn on the screen reader
2. navigate the home page via keyboard
3. Notice that individual items are read aloud, but section headers are never announced when entering a new section

### Expected Behavior

When navigating to a new section like "Recently added" or "Categories", the screen reader should announce the title of the section like it does when navigating to a properly labeled ListBox

### OS Version

8.x (Circe)

### OS Architecture

amd64 (on most hardwares)

### Session Type

Secure Session (Wayland)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell

```
